server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeFiltersTests.java

```diff
@@ -295,6 +295,16 @@ public void testOnlyAttributeValueFilter() {
         assertThat(discoveryNodeFilters.isOnlyAttributeValueFilter(), is(discoveryNodeFilters.match(node)));
     }
 
+    public void testIpv6WithScopeIdFromSettingsMatchingIpv6WithCompressedScopeId() {
+        Settings settings = shuffleSettings(Settings.builder()
+            .put("xxx." + randomFrom("_ip", "_host_ip", "_publish_ip"), "fdbd:dc00:111:222:0:0:0:333")
+            .build());
+        DiscoveryNodeFilters filters = buildFromSettings(OR, "xxx.", settings);
+
+        DiscoveryNode node = new DiscoveryNode("", "", "", "", "fdbd:dc00:111:222::333", localAddress, emptyMap(), emptySet(), null);
```

Comment:
Does this new mechanism also normalise IPv4 addresses by removing leading zeroes? That sounds desirable, it'd be good to have a test for this too to document this fact too.

Reply:
No, we don't support parsing IPv4 addresses with leading zeroes: https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/common/network/InetAddresses.java#L114

Reply:
Ah excellent thanks for checking. Just IPv6 then.